test/quantization/test_workflow_module.py

```diff
@@ -482,6 +482,29 @@ def test_save_load_state_dict_script(self):
             # Verify that state_dict matches exactly with original one.
             self.assertEqual(scripted.state_dict(), scripted_2.state_dict())
 
+
+    @unittest.skipIf(not TEST_MULTIGPU, "multi-GPU not supported")
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    def test_observer_qparams_respects_device_affinity(self):
+        """
+        Ensure that the scale and zero_point returned by the observer
+        are on the same device as the input tensor.
+        """
+        observerList = [MinMaxObserver(),
+                        MovingAverageMinMaxObserver(),
+                        PerChannelMinMaxObserver(),
+                        MovingAveragePerChannelMinMaxObserver()]
```

Comment:
hmm, looking at this test closely, I think this should fail unless the user moves the observers to CUDA explicitly.  I think the invariant to shoot for could be "observers should create tensors on the right device" and "observers should ensure that an input tensor has the same device as its internal buffers".    Thoughts?